### ISSUE TITLE
N-Body + CBS driver issue

### ISSUE THREAD

**Issue by PeterKraus** [psi4/psi4#1649]
The following input:

```
E, wfn = energy(cbs, cbs_metadata=[{"wfn":"hf", "basis": "cc-pwcv[tq]z-pp"},
                                     {"wfn":"mp2", "basis": "cc-pwcvtz-pp"}],
                  molecule=dimer,
                  return_wfn=True, dertype=0,
                  bsse_type="cp")
```

crashes with `UnboundLocalError: local variable 'bsse' referenced before assignment`. This is related to the logic on line 665:

https://github.com/psi4/psi4/blob/8e9d6888d722d071a9043bd8707ded67df81d7d3/psi4/driver/driver_nbody.py#L661-L668

The first (and only) assignment into `bsse` comes after, on line 681:

https://github.com/psi4/psi4/blob/8e9d6888d722d071a9043bd8707ded67df81d7d3/psi4/driver/driver_nbody.py#L680-L685

I'm not sure what this logic is doing exactly, but I'm guessing just switching the order of the blocks might do the trick.

**Comment by PeterKraus** [comment]
As @hokru pointed out, this is because I forgot to define fragments in my system. Perhaps we should check for that in the n-body driver somehow and print a better error message?

**Comment by loriab** [comment]
Agreed, checking that the sys has at least as many bodies as the nbody requests would be a good sanity check. Mol.nfragments() is the fn, I think. run_sapt in proc.py does something similar.

**Comment by dgasmith** [comment]
Probably something best here:
https://github.com/psi4/psi4/blob/8e9d6888d722d071a9043bd8707ded67df81d7d3/psi4/driver/driver_nbody.py#L239-L242

```python
if metadata['molecule'].nfragments() == 1:
    raise PsiExcetion()
elif metadata['max_nbody'] == -1:
...
```

**Comment by JonathonMisiewicz** [comment]
Meant to be closed in #1714, but typo happened. Oops.
